themes/default/content/blog/tour-pulumi-equinix-provider/index.md

```diff
@@ -0,0 +1,165 @@
+
+{{% chooser language "typescript,python" / %}}
+
+{{% choosable language typescript %}}
+
+```typescript
+const k8sProvider = new k8s.Provider("k8s-provider", {
+    kubeconfig: kubeconfig,
+});
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+k8s_provider = k8s.Provider(
+    "k8s-provider",
+    kubeconfig=kubeconfig
+)
+```
+
+{{% /choosable %}}
+
+You can then add resources to the cluster by specifying your explicit provider as a [resource option](https://www.pulumi.com/docs/concepts/options/):
+
+{{% chooser language "typescript,python" / %}}
+
+{{% choosable language typescript %}}
+
+```typescript
+const nginxDeployment = new k8s.apps.v1.Deployment("nginx-deployment", {
+    metadata: {
+        // ...
+    },
+    spec: {
+        // ...
+    },
+}, { provider: k8sProvider });
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+deployment = k8s.apps.v1.Deployment(
+    "nginx-deployment",
+    metadata={
+        # ...
+    },
+    spec={
+        # ...
+    },
+    opts=pulumi.ResourceOptions(
+        provider=k8s_provider
+    )
+)
+```
+
+{{% /choosable %}}
+
+## Adding storage
+
+Our cluster as configured will not be able to run stateful workloads due to the lack of a [StorageClass](https://kubernetes.io/docs/concepts/storage/storage-classes/) from which [PersistentVolumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) can be provisioned. Popular storage options for bare-metal/edge Kubernetes clusters include [Portworx](https://www.purestorage.com/products/cloud-native-applications/portworx.html), [Longhorn](https://longhorn.io/), [Rook](https://rook.io/), or (for non-production scenarios as data loss is likely if a node goes down) [NFS](https://kubernetes.io/docs/concepts/storage/storage-classes/#nfs).
+
+## Learning more
+
+## Conclusion
+
+The Pulumi Equinix provider offers developers an intuitive and efficient way to interact with Equinix resources. By combining the power of Pulumi's infrastructure-as-code tooling and ecosystem along with the utility of Equinix's service offerings, you can create and manage networking and bare metal compute resources using a single tool, freeing practitioners from manual configuration so they can focus on value-driving innovation.
+
+{{% notes type="info" %}}
+If you enjoyed this content we'd love it if you joined us (the authors of this post) on September 13, 2023 for a live hour-long workshop on [Deploying a Kubernetes Cluster on Equinix Metal](https://www.pulumi.com/resources/deploying-a-kubernetes-cluster-on-equinix-metal/).
```

Comment:
Can we move this to the beginning with the following CTA?

> Join Pulumi and Equinix on September 13, 2023, for a live workshop: [Deploying a Kubernetes Cluster on Equinix Metal](https://www.pulumi.com/resources/deploying-a-kubernetes-cluster-on-equinix-metal/).

Reply:
Done.